packages/@aws-cdk/aws-route53-patterns/lib/website-redirect.ts

```diff
@@ -0,0 +1,73 @@
+import { DnsValidatedCertificate, ICertificate } from '@aws-cdk/aws-certificatemanager';
+import { CloudFrontWebDistribution, OriginProtocolPolicy, PriceClass, ViewerProtocolPolicy } from '@aws-cdk/aws-cloudfront';
+import { ARecord, IHostedZone, RecordTarget } from '@aws-cdk/aws-route53';
+import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets';
+import { Bucket, RedirectProtocol } from '@aws-cdk/aws-s3';
+import { Construct, Fn, RemovalPolicy } from '@aws-cdk/core';
+
+export interface HTTPSRedirectProps {
+  /**
+   * HostedZone of the domain
+   */
+  readonly zone: IHostedZone;
+  /**
+   * The redirect target
+   */
+  readonly redirectTarget: string;
+  /**
+   * The domain name
+   *
+   * @default - the domain name of the zone
+   */
+  readonly domainName?: string;
+  /**
+   * The ARN of the certificate; Has to be in us-east-1
+   *
+   * @default - create a new certificate in us-east-1
+   */
+  readonly certificate?: ICertificate;
+}
+
+export class HTTPSRedirect extends Construct {
+  constructor(scope: Construct, id: string, props: HTTPSRedirectProps) {
+    super(scope, id);
+
+    const domainName = props.domainName || props.zone.zoneName;
+
+    const redirectCertArn = props.certificate ? props.certificate.certificateArn : new DnsValidatedCertificate(this, 'RedirectCertificate', {
+      domainName,
+      hostedZone: props.zone,
+      region: 'us-east-1',
+    }).certificateArn;
+
+    const redirectBucket = new Bucket(this, 'RedirectBucket', {
+      websiteRedirect: {
+        hostName: props.redirectTarget,
+        protocol: RedirectProtocol.HTTPS,
+      },
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+    const redirectDist = new CloudFrontWebDistribution(this, 'RedirectDistribution', {
+      originConfigs: [{
+        behaviors: [{ isDefaultBehavior: true }],
+        customOriginSource: {
+          domainName: Fn.select(2, Fn.split('/', redirectBucket.bucketWebsiteUrl)),
```

Comment:
Are you sure there isn’t an attribute for this already?

Reply:
I presume that this is to remove the `http://` from the website url attribute

Reply:
Wouldn't that be `redirectBucket.bucketDomainName`?

Reply:
`bucketDomainName` points to the REST API endpoint, not the website endpoint. The REST API endpoint doesn't support redirects.

`mystack-mybucket-kdwwxmddtr2g.s3.amazonaws.com` vs `http://mystack-mybucket-kdwwxmddtr2g.s3-website-us-east-2.amazonaws.com/`

https://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteEndpoints.html#WebsiteRestEndpointDiff

@hoegertn may be add a comment explaining this line.

Reply:
the bucketDomainName is the name for S3 operations but not the website domain

Reply:
Should we simply add another attribute to `s3.Bucket`? Sounds like something others can benefit from...

Reply:
@hoegertn care to raise an issue (and make sure to reference this usage so we can track it down).